skimage/measure/fit.py

```diff
@@ -610,19 +612,14 @@ def _dynamic_max_trials(n_inliers, n_samples, min_samples, probability):
     trials : int
         Number of trials.
     """
+    if probability == 0:
+        return 0
     if n_inliers == 0:
         return np.inf
-
-    if probability == 1:
-        return np.inf
-
-    if n_inliers == n_samples:
-        return 1
```

Comment:
I don't know if it worth it, but shouldn't we also keep this special cases?

Reply:
I think it's necessary.
By the way, if the default value of probability is 1, the effect of dynamic maximum decreases, so what do you think about adding that code and adjusting the default value of probability from 1 to 0.99.

Reply:
Changing the default value is a breaking change and needs starting a deprecation cycle. We need more @scikit-image/core inputs to take this decision.

Reply:
Can you explain what depression cycle is? I'm Sorry :(
Additionally, do you think it's better to proceed with that process? Or do you think it's better to just add the code above and maintain the default value?

Reply:
You can find in the [contribution guide](https://scikit-image.org/docs/dev/contribute.html#deprecation-cycle) the strategy that `skimage` adopts to manage any behavior change in the library.
I have no strong opinion about what the default value should be, but I see that `sklearn` uses 0.99 as default.

Reply:
Thank you. I checked contribution guide and sklearn's PR when _dynamic_max_trials were first implemented.
If the results when _dynamic_max_trials are applied are not different from the existing results and the speed increases, I think it is okay to set the stop_probability to the default value of 0.99 which is large enough but not 1.

`We need more @scikit-image/core inputs to take this decision.`
-> Does this mean we need to get opinions from other maintainers?

Reply:
> -> Does this mean we need to get opinions from other maintainers?

Yes, that's what I meant :wink:

Reply:
Thank you. Is there anything I need to do to get an opinion? Is it enough to just wait?
Because I can't stand my commit being merged. It's so excited😆

Reply:
We simply need to wait now: core devs are already notified, just give them some time to answer :wink:

Reply:
Okay, Thanks a lot. 🙂

Reply:
I think this is fine to put in without deprecation.

Reply:
@stefanv
So you're saying we should add the code above to _dynamic_max_trials and keep the default stop_probability value of ransac() at 1, right?

Reply:
Why does sklearn use 0.99?

Reply:
In my opinion, if it is set to 1, the dynamic maximum is not calculated, so it seems that the value is set to 0.99, so that the dynamic maximum is calculated without affecting the result as much as possible.
In other words, I think this is just the difference in decision whether to use dynamic maximum or not.
What do you think about it @stefanv ?

Reply:
@stefanv @rfezzani 
The default value for the probability of skimage is set to 1. Therefore, the existing skimage code always returns the infinite number of trials. I think this is an inefficient way.
Since the commonly used value is 0.99, it is reasonable to set the default to 0.99 **if the user does not want to pay particular attention** to probability. In the following paper, 0.99 is used as the value of RANSAC's confidence probability. [FestGPU: a framework for fast robust estimation on GPU](https://www.researchgate.net/figure/The-influence-of-the-RANSAC-parameters-to-the-number-of-iterations-p-099-For-larger_fig6_268124277)
And like @stefanv said, I think the fact that the default value of probability in sklearn is 0.99 can be the basis for changing the default value of skimage.

Reply:
And from a different point of view, even if the default value is not modified, I think it will have a similar effect to the above discussion if the code compares the value that goes into the log as below with epsilon.
```
nom = max(_EPSILON, 1 - probability)
denom = max(_EPSILON, 1 - inlier_ratio ** min_samples)
return float(np.ceil(np.log(nom) / np.log(denom)))
```

In the same context, we don't need the code below, do we?
```
if probability == 1:
    return np.inf

if n_inliers == n_samples:
    return 1
```

Reply:
I think that we can remove the handling of these special cases.

Reply:
The test for `probability == 1` is necessary for the tests to pass :)

Reply:
@rfezzani Oh, I see. I modified it!